automation/taskcluster/androidTest/parse-ui-test.py

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/python3
+
+from __future__ import print_function
+
+import sys
+import argparse
+from pathlib import Path
+import json
+import yaml
+
+def parse_args(cmdln_args):
+    parser = argparse.ArgumentParser(description="Parse UI test logs an results")
+    parser.add_argument(
+        "--output-md",
+        type=argparse.FileType("w", encoding="utf-8"),
+        help="Output markdown file.",
+        required=True,
+    )
+    parser.add_argument(
+        "--log",
+        type=argparse.FileType("r", encoding="utf-8"),
+        help="Log output of flank.",
+        required=True,
+    )
+    parser.add_argument(
+        "--results", type=Path, help="Directory containing flank results", required=True
+    )
+    parser.add_argument(
+        "--exit-code", type=int, help="Exit code of flank.", required=True
+    )
+    parser.add_argument("--device-type", help="Type of device ", required=True)
+    return parser.parse_args(args=cmdln_args)
+
+
+def extract_android_args(log):
+    return yaml.safe_load(log.split("AndroidArgs\n")[1].split("RunTests\n")[0])
+
+
+def main():
+    args = parse_args(sys.argv[1:])
+
+    log = args.log.read()
+    matrix_ids = json.loads(args.results.joinpath("matrix_ids.json").read_text())
+    #with args.results.joinpath("flank.yml") as f:
+    #    flank_config = yaml.safe_load(f)
+
+    android_args = extract_android_args(log)
+
+    print = args.output_md.write
```

Comment:
OK, makes sense.  I'll take a first pass at what you propose.  Let me know if it looks correct.

Reply:
I'm sorry I wasn't super explicit and my original code didn't work. I meant something closer to this:

```py
matrix_results_per_id = json.loads(args.results.joinpath("matrix_ids.json").read_text())
android_args = extract_android_args(log)
markdown = format_test_results_to_markdown(android_args["gcloud"]["device"], matrix_results_per_id):
args.output_md.write(markdown)


def format_test_results_to_markdown(devices, matrix_results_per_id):
    markdown_lines = [
        "# Devices",
        devices,
        "# Results",
        "| matrix | result | logs |",
        "| --- | --- | --- |",
    ]

    markdown_lines.extend([
        "| {matrixId} | {outcome} | [logs]({webLink}) |".format(**matrix_result)
        for matrix_results in matrix_results_per_id.values()
    ])

    return "\n".join(markdown_lines)
```

Does it look better to you?

Reply:
> @JohanLorenzo once we have this effectively rolled out on A-C, I'll update fenix to reflect changes we've made here as well as open a PR on reference-browser to enable the same. hopefully, you won't mind giving those a review too. I realize it will be tricky here because we'll need to coordinate re-activating making UI tests build blocking as they were before #7617 disabled that. I don't think UI tests are build blocking on R-B yet, but we'd like to have this be standard across all mobile projects and I don't think that would be hard to do for R-B since there is little activity there

@JohanLorenzo chatted a bit with @pocmo on this.  Sounds like how the UI tests will be triggered is changing sometime soon so lets not do anything there for now.  Can we simply land these job logging changes?  My only concern is since they won't launch on git push we won't know if this will break things until git merge.  Is there a safer way to land this?